var/spack/repos/builtin/packages/py-setuptools-scm/package.py

```diff
@@ -10,11 +10,12 @@ class PySetuptoolsScm(PythonPackage):
     """The blessed package to manage your versions by scm tags."""
 
     homepage = "https://github.com/pypa/setuptools_scm"
-    pypi = "setuptools_scm/setuptools_scm-4.1.2.tar.gz"
+    pypi = "setuptools-scm/setuptools-scm-8.0.4.tar.gz"
```

Comment:
This breaks older version installs since the URL is not derived correctly.

Reply:
Not canonical spack, but this does the job:
```py
    def url_for_version(self, version):
        if version >= Version(8):
            return super().url_for_version(version).replace("_", "-")
        else:
            return super().url_for_version(version)
```

Reply:
"You could call super once and store it in a variable." https://spackpm.slack.com/archives/C5VL7V81G/p1714549700619169?thread_ts=1714517377.256679&cid=C5VL7V81G